GroundDBMemoryPool/src/rdma/rdma_client/connect.cc

```diff
@@ -56,7 +57,45 @@ struct resources *connectServer(
     }
     fprintf(stdout, "Connection verified\n");
     // Remove until here
+
+    auto pat_req_buf = new request_buffer(1<<20);
+    init_pat_access_on_client(res, pat_req_buf, server_name, tcp_port, ib_port);
     return res;
 }
 
+void init_pat_access_on_client(
+        struct resources* res,
+        struct request_buffer* pat_req_buf,
+        const char *server_name, /* server host name */
+        const int tcp_port,     /* server TCP port */
+        const int ib_port       /* server IB port */
+){
+    struct memory_region *pat_mr = nullptr;
+    if (register_mr(pat_mr, res, (char*)pat_req_buf->content_, pat_req_buf->size_)){
+        fprintf(stderr, "failed to register memory regions\n");
+        exit(1);
+    }
+    struct connection *pat_conn = nullptr;
+    if (connect_qp(pat_conn, res, pat_mr, server_name, tcp_port, -1, ib_port))
+    {
+        fprintf(stderr, "failed to connect QPs\n");
+        exit(1);
+    }
+    sock_sync_data(pat_conn);
+}
+uintptr_t get_page_address_from_mempool(
+        const struct resources *res,
+        const struct memory_region *mr,
+        const struct connection *conn,
+        KeyType page_id
+){
+    post_receive(res, mr, conn, 0, sizeof(uintptr_t));
+    *(KeyType*)mr->buf = page_id;
+    post_send(res, mr, conn, IBV_WR_SEND, 0, sizeof(KeyType));
+    poll_completion(res, conn);
+    poll_completion(res, conn);
```

Comment:
Duplicate line

Reply:
It is actually not. They are for the `post_send` and `post_receive` respectively